### ISSUE TITLE
Documentation for git_sort_t needs clarification

### ISSUE THREAD

**Issue by swisspol** [libgit2/libgit2#2765]
Assuming we have this history where commits were created in this numerical order with 1s difference between them:

```
  0---1----2----4----7 (master)
       \         \
        3----5----6----8--- (topic)
```

Libgit2 commit walker API outputs the following depending on the sorting mode:
- `GIT_SORT_NONE`: 7, 4, 2, 1, 0, 8, 6, 5, 3
- `GIT_SORT_TIME`: 8, 7, 6, 5, 4, 3, 2, 1, 0
- `GIT_SORT_TOPOLOGICAL`: 8, 6, 5, 3, 7, 4, 2, 1, 0
- `GIT_SORT_TOPOLOGICAL | GIT_SORT_TIME`: 7, 8, 6, 4, 2, 5, 3, 1, 0

The doc for `GIT_SORT_TOPOLOGICAL` however says that parents will come before children which is not the case obviously:

```
    /**
     * Sort the repository contents in topological order
     * (parents before children); this sorting mode
     * can be combined with time sorting.
     */
    GIT_SORT_TOPOLOGICAL = 1 << 0,
```

The doc don't explain either how exactly combining `GIT_SORT_TOPOLOGICAL | GIT_SORT_TIME` works: it just says they can be combined. For instance which mode takes priority over the other? The test above just adds confusion as both `GIT_SORT_TIME` and `GIT_SORT_TOPOLOGICAL` put commit 8 first but combining both modes put commit 7 first.

For reference, here's how the Git log command help explains the available sorting modes which is easier to understand and less ambiguous IMO:

```
Commit Ordering
By default, the commits are shown in reverse chronological order.

--date-order
Show no parents before all of its children are shown, but otherwise show commits in the commit timestamp order.

--author-date-order
Show no parents before all of its children are shown, but otherwise show commits in the author timestamp order.

--topo-order
Show no parents before all of its children are shown, and avoid showing commits on multiple lines of history intermixed.

--reverse
Output the commits in reverse order.
```

**Comment by swisspol** [comment]
Anyone?

**Comment by ethomson** [comment]
Fixed via #4811
